hitmap.go

```diff
@@ -49,6 +50,7 @@ func (hits *hitMap) expireLimits() {
 	for {
 		hitMapCopy := hits.getMap()
 		for _, ipr := range hitMapCopy {
+			// TODO address concurrent read
```

Comment:
Oh, its not a deep copy. Could potentially be an issue if it is simultaneously set and read, but pretty low risk. Wouldn't hurt to add a mutex to ipr if concerned.